azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/impl/EventDataImpl.java

```diff
@@ -258,4 +258,17 @@ private void readObject(ObjectInputStream in) throws IOException, ClassNotFoundE
             this.bodyData = new Binary(data, 0, length);
         }
     }
+
+    @Override
+    public int compareTo(EventData other) {
+        long seqThis = this.getSystemProperties().getSequenceNumber();
+        long seqOther = other.getSystemProperties().getSequenceNumber();
+
+        if(seqThis < seqOther) {
+          return -1;
+        } else if (seqThis > seqOther){
+          return 1;
+        }
+        return 0;
```

Comment:
You could use Long.comapre() method to simplify it - Long.compare(this.getSystemProperties().getSequenceNumber(), other.getSystemProperties().getSequenceNumber());

Reply:
Perfect suggestion, I've updated the code.